src/battle/overlay_12_0224E4FC.c

```diff
@@ -3864,3 +3864,689 @@ int ov12_02253194(BattleSystem *bsys, BATTLECONTEXT *ctx) {
+            }
+            break;
+        case HOLD_EFFECT_BRN_RESTORE: //rawst berry
+            if (ctx->battleMons[battlerId].status & STATUS_BURN) {
+                script = 202;
+                ret = TRUE;
+            }
+            break;
+        case HOLD_EFFECT_FRZ_RESTORE: //aspear berry
+            if (ctx->battleMons[battlerId].status & STATUS_FREEZE) {
+                script = 203;
+                ret = TRUE;
+            }
+            break;
+        case HOLD_EFFECT_PP_RESTORE: //leppa berry
+            int index;
+            for (index = 0; index < LEARNED_MOVES_MAX; index++) {
+                if (ctx->battleMons[battlerId].moves[index] && !ctx->battleMons[battlerId].movePPCur[index]) {
+                    break;
+                }
+            }
+            if (index != LEARNED_MOVES_MAX) {
+                AddBattlerVar(&ctx->battleMons[battlerId], BMON_DATA_MOVE1PP + index, boost);
+                CopyBattleMonToPartyMon(bsys, ctx, battlerId);
+                ctx->moveWork = ctx->battleMons[battlerId].moves[index];
+                script = 204;
+                ret = TRUE;
+            }
+            break;
+        case HOLD_EFFECT_CONFUSE_RESTORE: //persim berry
+            if (ctx->battleMons[battlerId].status2 & STATUS2_CONFUSION) {
+                script = 205;
+                ret = TRUE;
+            }
+            break;
+        case HOLD_EFFECT_STATUS_RESTORE: //lum berry
+            if ((ctx->battleMons[battlerId].status & STATUS_ALL) || (ctx->battleMons[battlerId].status2 & STATUS2_CONFUSION)) {
+                if (ctx->battleMons[battlerId].status & STATUS_PARALYSIS) {
+                    script = 199;
+                }
+                if (ctx->battleMons[battlerId].status & STATUS_SLEEP) {
+                    script = 200;
+                }
+                if (ctx->battleMons[battlerId].status & STATUS_POISON_ALL) {
+                    script = 201;
+                }
+                if (ctx->battleMons[battlerId].status & STATUS_BURN) {
+                    script = 202;
+                }
+                if (ctx->battleMons[battlerId].status & STATUS_FREEZE) {
+                    script = 203;
+                }
+                if (ctx->battleMons[battlerId].status2 & STATUS2_CONFUSION) {
+                    script = 205;
+                }
+                if ((ctx->battleMons[battlerId].status & STATUS_ALL) && (ctx->battleMons[battlerId].status2 & STATUS2_CONFUSION)) {
+                    script = 206;
+                }
+                ret = TRUE;
+            }
+            break;
+        case HOLD_EFFECT_HP_RESTORE_SPICY: //figy berry
+            if (ctx->battleMons[battlerId].hp <= ctx->battleMons[battlerId].maxHp / 2) {
+                ctx->hpCalcWork = DamageDivide(ctx->battleMons[battlerId].maxHp, boost);
+                ctx->msgWork = 0;
+                if (GetFlavorPreferenceFromPID(ctx->battleMons[battlerId].personality, 0) == -1) {
```

Comment:
personality enum/consts?

Reply:
huh? you mean the 0 in the function?

Reply:
yeah, that varies with each personality difference

Reply:
oh I think those are flavor consts, let me see if we have them and I'll add them if not